### PR TITLE
[export-csv] Improve management when data are incomplete (#3617)

### LINKED ISSUE #3617: Change behavior when an export fails to export (EXport CSV)
## Description

When a EXport CSV fails to export, currently, the whole export fails. 
It would be great to be able to export whatever can be exported, and gracefully handle errors the same way we do for import, by logging all the errors instead of failing the whole export.

This is classified as a bugfix since it would solve: https://github.com/OpenCTI-Platform/opencti/issues/10008

Reproduction steps on https://testing.octi.staging.filigran.io/dashboard/threats/intrusion_sets/cbdbff91-23bb-44ea-b46d-43845f1c22ea/knowledge/indicators

- export all indicators in the knowledge view as csv file.

### PR DESCRIPTION
**Pull request by aHenryJard**
### Proposed changes

* Add a global try/cath per csv line, to allow exporting csv event if one line has issue
* Add some test on value that are lists inside, to avoid crashing on null/None values

Error log on catch looks like:
```
{"timestamp": "2025-03-31T12:55:02.411462Z", "level": "WARNING", "name": "ExportFileCsv", "message": "Error with csv input data, one line cannot be exported.sequence item 0: expected str instance, NoneType found"}
```

On opencti connector view when there is an error:
image

### Related issues

* closes #3617 

### Checklist

- [ ] I consider the submitted work as finished
- [ ] I have signed my commits using GPG key.
- [ ] I tested the code for its functionality using different use cases
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: internal-export-file/export-file-csv/src/export-file-csv.py [hunk 1]
```diff
@@ -40,53 +40,68 @@ def export_dict_list_to_csv(self, data):
...
+                                elif "value" in r:
+                                    if r["value"] is not None:
+                                        rrow.append(r["value"])
+                                    else:
+                                        rrow.append("")
+                                elif "observable_value" in r:
+                                    if r["observable_value"] is not None:
+                                        rrow.append(r["observable_value"])
+                                    else:
+                                        rrow.append("")
```
Comment (flavienSindou) [inline internal-export-file/export-file-csv/src/export-file-csv.py:90]: Thank you for this explicit and comprehensive if/else clause for handling null values.

Just a tip: you could also have used the pythonic syntax:
 ```python
r["whatever"] or ""
```

File: internal-export-file/export-file-csv/src/export-file-csv.py [hunk 2]
```diff
@@ -40,53 +40,68 @@ def export_dict_list_to_csv(self, data):
...
+                        else:
+                            row.append("")
                     else:
                         row.append("")
-                else:
-                    row.append("")
-            csv_data.append(row)
+                csv_data.append(row)
+            except Exception as err:
+                self.helper.log_error("ERROR with csv input data." + str(err))
```
Comment (flavienSindou) [inline internal-export-file/export-file-csv/src/export-file-csv.py]: Good approach to allow users to retrieve their CSV even if a single row is corrupted! 👍

We now have updated Product team guidelines regarding logging levels:

 * Info: Used when we know the connector might not handle a certain data type, and this is explicitly documented.
 * Warning: Used when a packet to be processed is absent, corrupted, etc., and is skipped while the process continues.
* Error: Used when the connector is broken due to a code error.

Could you update the log level accordingly?

```suggestion
            except Exception as err:
                self.helper.connector_logger.warning("ERROR with csv input data." + str(err))
```

Additionally, you might consider (to be discussed with the Product team) informing the user directly in the OCTI platform when the export is not fully successful. This could be done using the _process_message return value. However, implementing this would require a more significant change to store an error list.

For example

```python
class ExportFileCsv:
    def __init__(self):
        ...
        self.errors: list[Exception] = [] # error holder to be reset before each new process

    def export_dict_list_to_csv
        ...     
            except Exception as err:
                self.helper.connector_logger.warning("ERROR with csv input data." + str(err))
                errors.append(err)
        ...
    ...
    def _process_message(self, data):
        self.errors = [] # reset before launching main process
        ...
        if errors:
            msg = f"Some corrupted values were not processed ({len(self.errors)}). See connector logs for details."
            return msg
        return "Export done"
```

**Review by flavienSindou — CHANGES_REQUESTED** [review]
Thanks for this improvement and the way you’ve handled CSV row corruption to ensure users still get their data. I’ve requested a small change regarding the log level to align with our new Product team guidelines. Also, you might want to check with them about notifying users directly in the OCTI platform.

File: internal-export-file/export-file-csv/src/export-file-csv.py (hunk 2 above)
Comment (aHenryJard) [reply internal-export-file/export-file-csv/src/export-file-csv.py]: I love that array of errors, I will give a try

**Review by flavienSindou — APPROVED** [review]
Great work ! Thank you for this contribution !
